agent/provider/src/market/negotiator/builtin/expiration.rs

```diff
@@ -102,18 +102,24 @@ impl NegotiatorComponent for LimitExpiration {
         let max_expiration = now + max_expiration_delta;
         let min_expiration = now + self.min_expiration;
 
-        if req_expiration > max_expiration || req_expiration < min_expiration {
+        let too_late = req_expiration < min_expiration;
+        let too_soon = req_expiration > max_expiration;
+        if too_soon || too_late {
             log::info!(
                 "Negotiator: Reject proposal [{}] due to expiration limits.",
                 demand.agreement_id
             );
+
             return Ok(NegotiationResult::Reject {
-                reason: Some(Reason::new(format!(
-                    "Proposal expires at: {} which is less than {} or more than {} from now",
-                    req_expiration,
-                    self.min_expiration.display(),
-                    max_expiration_delta.display()
-                ))),
+                reason: Some(reason_with_extra(
+                    format!(
+                        "Proposal expires at: {} which is less than {} or more than {} from now",
+                        req_expiration,
+                        self.min_expiration.display(),
+                        max_expiration_delta.display()
+                    ),
+                    serde_json::json!({ "golem.proposal.rejection.is-final": too_late }), // when it's too soon we could try later
```

Comment:
Is this comment true?
If expiration is to close to `now`, Provider will never pickup this Proposal

Reply:
If demand came too_soon, then in the future it might be inside the acceptance window

Reply:
But it doesn't matter when Demand came. Expiration timestamp will always be closer

Reply:
https://github.com/golemfactory/yagna/blob/9928a2f44fe69aabb3142c5139e423e03668a169/agent/provider/src/market/negotiator/builtin/expiration.rs#L100-L105

`too_soon` means that requestor expiration is bigger than max expiration. As far as I understand it will become smaller and smaller with passing time. Am I missing something?